src/include/distributed/commands.h

```diff
@@ -24,6 +24,12 @@
 
 extern bool AddAllLocalTablesToMetadata;
 
+/*
+ * TODO: should we require AddAllLocalTablesToMetadata being false
+ *       to enable this?
+ */
```

Comment:
.

Reply:
Maybe no, but we should check for schema-based sharding logic before citus managed tables branch (if / else if) to avoid going through table conversions twice.